d3servercheck.ps1

```diff
@@ -1,70 +1,52 @@
-function Write-Server-Result($tcpconns) {
+function Write-ServerResult($tcpconns) {
     if (!$tcpconns) {
         return
     }
-
-    foreach ($tcpconn in $tcpconns) {
-        Write-Host "$($tcpconn.RemoteAddress) (" -NoNewline
-        if ($is_good_ip.ContainsKey($tcpconn.RemoteAddress)) {
-            if ($is_good_ip[$tcpconn.RemoteAddress]) {
-                Write-Host "GOOD" -NoNewline
+    $Unique_Addresses = $tcpconns | Sort-object -property RemoteAddress -unique
+    write-host "Connections:"
+    foreach ($address in $Unique_Addresses) {
+        if ($is_good_ip.ContainsKey($address.RemoteAddress)) {
+            if ($is_good_ip[$address.RemoteAddress]) {
+                $Result = "GOOD"
             } else {
-                Write-Host "BAD" -NoNewline
+                $Result =  "BAD" 
             }
-        } else {
-            Write-Host "UNKNOWN" -NoNewline
         }
-
-        Write-Host ") " -NoNewline
+        else{
+            $Result = 'Unknown'
+        }
+        
+        $Message = '{0} : ({1})' -f $address.remoteaddress,$Result
+        Write-Host "$message"
     }
+    write-host "`r"
 }
 
 Write-Host "Starting d3servercheck"
 
 $csv = Import-Csv -Path servers.csv
 
+try {
+    $Diablo3_Process = get-process -name 'Diablo III64' -ErrorAction Stop
+}
+Catch{
+    write-error "Diablo 3 is not running. Exiting script..."
+    exit 1
```

Comment:
This changes the script's semantic. In the README I state that the script can be started at any time and will check D3 connections dynamically, but suddenly the script fails if D3 is not running. If we would implement this the documentation has to change as well.

However I'd prefer if the script kept it's dynamicness, as it increases the ease-of-use, as described in the previous comment.